core/state.go

```diff
@@ -274,6 +282,11 @@ func (s *State) updateContracts(stateTrie *trie.Trie, blockNumber uint64, diff *
 		}
 	}
 
+	err := s.checkAndDeploySpecialContracts(stateTrie, blockNumber, diff.StorageDiffs)
+	if err != nil {
+		return err
+	}
+
```

Comment:
We should do this in reverse in `State.Revert`

Reply:
Done